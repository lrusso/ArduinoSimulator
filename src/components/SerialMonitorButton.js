import React from "react"
import { t } from "../utils/service"

const SerialMonitorButton = ({ sendSerialData, simulatorRunning }) => {
  return (
    <input
      type="submit"
      style={{ ...styles.button, color: simulatorRunning ? "black" : "gray" }}
      value={t("SEND")}
      disabled={!simulatorRunning}
      onClick={sendSerialData}
    />
  )
}

const styles = {
  button: {
    borderRadius: 0,
    marginLeft: "2px",
    fontFamily: "Arial",
    fontSize: "13px",
    border: "1px solid silver",
    backgroundColor: "#D9D9D9",
    height: "28px",
    width: "99px",
    marginRight: "5px",
    WebkitAppearance: "none",
  },
}

export default SerialMonitorButton
