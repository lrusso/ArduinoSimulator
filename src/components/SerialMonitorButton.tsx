import React from "react"
import { t } from "../utils/languages"

interface SerialMonitorButtonProps {
  sendSerialData: () => void
  simulatorRunning: boolean
}

const SerialMonitorButton = ({
  sendSerialData,
  simulatorRunning,
}: SerialMonitorButtonProps) => {
  return (
    <input
      type="submit"
     className="serial_send"
      value={t("SEND")}
      disabled={!simulatorRunning}
      onClick={sendSerialData}
    />
  )
}


export default SerialMonitorButton
