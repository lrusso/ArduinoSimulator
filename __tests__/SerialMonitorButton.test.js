import React from "react"
import renderer from "react-test-renderer"
import SerialMonitorButton from "../src/components/SerialMonitorButton"

describe("SerialMonitorButton", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<SerialMonitorButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
