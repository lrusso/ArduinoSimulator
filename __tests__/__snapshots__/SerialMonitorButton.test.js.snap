// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerialMonitorButton Rendering 1 1`] = `
<input
  disabled={true}
  style={
    {
      "WebkitAppearance": "none",
      "backgroundColor": "#D9D9D9",
      "border": "1px solid silver",
      "borderRadius": 0,
      "color": "gray",
      "fontFamily": "Arial",
      "fontSize": "13px",
      "height": "28px",
      "marginLeft": "2px",
      "marginRight": "5px",
      "width": "99px",
    }
  }
  type="submit"
  value="SEND"
/>
`;
